src/submodule.c

```diff
@@ -688,6 +691,34 @@ int git_submodule_add_setup(
 		goto cleanup;
 	}
 
+	/* get the index for the repo */
```

Comment:
It would be nice to put the following checks into a simple function. Something like `can_add_submodule(repo, path)`.